compiler/src/dmd/common/file.d

```diff
@@ -587,3 +594,116 @@ private auto ref fakePure(F)(scope F fun) pure
     mixin("alias PureFun = " ~ F.stringof ~ " pure;");
     return (cast(PureFun) fun)();
 }
+
+/***********************************
+ * Recursively search all the directories and files under dir_path
+ * for files that match one of the extensions in exts[].
+ * Pass the matches to sink.
+ * Params:
+ *      dir_path = root of directories to search
+ *      exts = array of filename extensions to match
+ *      recurse = go into subdirectories
+ *      filenameSink = accepts the resulting matches
+ * Returns:
+ *      true for failed to open the directory
+ */
+bool find_files(const char* dir_path, const char[][] exts, bool recurse, void delegate(const(char)[]) nothrow filenameSink)
+{
+    //printf("find_files() dir_path: %s\n", dir_path);
+    version (Windows)
+    {
+        char[MAX_PATH + 1] full_path = void;
+        snprintf(full_path.ptr, full_path.length, "%s\\*.*", dir_path);
+        //fprintf(stderr, "full_path: %s\n", full_path.ptr);
+
+        WIN32_FIND_DATAA ffd = void;
+        HANDLE hFind = FindFirstFileA(full_path.ptr, &ffd);
```

Comment:
This should use the unicode API, `FindFirstFileW`

Reply:
done